doc/book/app_server/luajit_memprof.rst

```diff
@@ -0,0 +1,524 @@
+
+LuaJIT memory profiler
+======================
+
+Starting from version :doc:`2.7.1 </release/2.7.1>`, Tarantool
+has the built-in module called ``misc.memprof`` that implements the LuaJIT memory
+profiler and the profile parser (further, *profiler*). The profiler provides
+a memory allocation report that helps analyse Lua code and find out the places
+that put the most pressure on the Lua garbage collector (GC).
+
+..  contents::
+    :local:
+    :depth: 2
+
+.. _profiler_usage:
+
+Working with profiler
+---------------------
+
+Usage of the profiler is two-fold:
+
+1.  :ref:`Collect <profiler_usage_get>` a binary profile of allocations,
+    reallocations, and deallocations in memory related to Lua VM
+    (further, *binary memory profile* or *binary profile* for short).
+2.  :ref:`Parse <profiler_usage_parse>` the binary profile collected and get
+    a human-readable profiling report.
+
+.. _profiler_usage_get:
+
+Collecting binary profile
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To collect a binary profile for a particular part of the Lua code,
+you need to place this part between two ``misc.memprof`` functions,
+namely, ``misc.memprof.start()`` and ``misc.memprof.stop()``, and then execute
+the code under Tarantool.
+
+Below is a chunk of simple Lua code named ``test.lua`` to illustrate this.
+
+.. _profiler_usage_example01:
+
+..  code-block:: lua
+    :linenos:
+
+    jit.off()
+    local str, err = misc.memprof.start("memprof_new.bin")
+    -- Lua doesn't create a new frame to call string.rep, and all allocations
+    -- are attributed not to the append() function but to the parent scope.
+    local function append(str, rep)
+        return string.rep(str, rep)
+    end
+
+    local t = {}
+    for i = 1, 1e5 do
+        -- table.insert is the built-in function and all corresponding
+        -- allocations are reported in the scope of the main chunk.
+        table.insert(t,
+            append('q', i)
+        )
+    end
+    local stp, err = misc.memprof.stop()
+
+..  note::
+
+    Usually, the information about allocations inside Lua built-ins are not really
+    useful for the developer. That's why if a Lua built-in function is called from
+    a Lua function, the profiler attributes all allocations to the Lua function.
+    Otherwise, this event is attributed to a C function.
+
+    Tail call optimization doesn't create a new call frame, so all allocations
+    inside the function called via the ``CALLT/CALLMT`` `bytecodes <http://wiki.luajit.org/Bytecode-2.0>`_
+    are attributed to its caller.
+
+    Example above illustrates these cases.
+
+Starting profiler in Lua code:
+
+..  code-block:: lua
+
+    local str, err = misc.memprof.start(FILENAME)
+
+where ``FILENAME`` is a name of the binary file where profile events are written.
+The writer for this function performs ``fwrite()`` for each call retrying
+in case of ``EINTR``. When the profiling is stopped, ``fclose()`` is called.
```

Comment:
I'm totally OK, if we move `fclose()` mentioning to `misc.memprof.stop()`.
Also, these mentions may be really redundant -- there is no singularity related to C standard functions in memprof API, so it can be omitted.

Reply:
as discussed, removed mentioning fclose and fwrite